lib/datamodel/graph.js

```diff
@@ -143,4 +147,60 @@ export class Graph {
 
     return shortestPath
   }
+
+  /**
+   * This returns an array of the graph's nodes in topologically sorted order.
+   * If the graph is not a DAG, no topological sorting is possible and an empty array is returned.
+   * 
+   * Implements Kahn's Algorithm.
+   * @return {array} An array of the graph's nodes in topologically sorted order.
+   */
+  topologicalSort() {
```

Comment:
This fine since it very much models Khan's Algorithm from https://www.geeksforgeeks.org/topological-sorting-indegree-based-solution.

This currently sorts the entire set of tables whereas the one in [BaseModelScanner](https://github.com/tair/morphobank/blob/master/app/lib/mb/BaseModelScanner.php#L171) has an ignore list and a defined list of tables to sort which is needed for model duplication. I'll approve but note that there is more work to exclude and include tables.

Reply:
@kenzley so later on would I have to update this topological sort method or create a different one that does that in the `BaseModelScanner` class?

Reply:
Most likely you'll have to create a different method to use in `BaseModelScanner`.

Note that the `getTopologicalDependentTables` method uses the topological sort algorithm but has modifications to ignore specific tables and ensure that all the tables specified are in the returned list. This method will return all tables.  Of course, you can do a set intersection to only return your wanted tables - but I'm not sure whether the order will be the same.  You'll have to have experiment to verify.